Doc/library/shelve.rst

```diff
@@ -41,13 +42,25 @@ lots of shared  sub-objects.  The keys are ordinary strings.
    determine which accessed entries are mutable, nor which ones were actually
    mutated).
 
+   By default, :mod:`shelve` uses :func:`pickle.dumps` and :func:`pickle.loads`
+   for serializing and deserializing. This can be changed by supplying
+   *serializer* and *deserializer*, respectively. The *serializer* argument
+   should be a function that takes an object and returns its representation
+   as a :term:`bytes-like object`; *deserializer* should be a function that
+   takes :class:`bytes` and returns the corresponding object.
```

Comment:
`pickle.loads` accepts bytes-like objects as well. If you accept arbitrary bytes objects as the `serializer` output, you should also test the various objects implementing the buffer protocol.